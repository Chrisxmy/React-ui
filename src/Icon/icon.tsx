import * as React from 'react';
import './icon.less';
import './importAll';

import { classNames } from '../utils/classNames';

export interface IconProps extends React.HTMLAttributes<HTMLElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> = ({
  className,
  style,
  name,
  ...restProps
}) => {
  return (
    <span {...restProps}>
      <svg className={classNames('m-icon', className)} style={style}>
        <use xlinkHref={`#icon-${name}`} />
      </svg>
    </span>
  );
};

export default Icon;
