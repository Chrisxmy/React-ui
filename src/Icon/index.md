---
nav:
  title: Components
  path: /components
group:
  title: Basic
  path: /basic
---

## Icon

Demo:

```tsx
import React from 'react';
import { Icon } from 'dumi';

export default () => {
  return (
    <>
      <Icon name="wechat" style={{ fill: 'green' }} />
      <Icon name="arrow-left" />
      <Icon name="arrow-right" />
      <Icon name="logo" style={{ width: '30px', height: '30px' }} />
      <Icon name="github" />
    </>
  );
};
```
