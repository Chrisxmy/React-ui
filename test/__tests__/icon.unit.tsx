import * as rendererer from 'react-test-renderer'
import * as React from 'react'
import Icon from '../../lib/icon/icon'
import { mount } from 'enzyme'

describe('icon', () => {

    it('render successfully', () => {
        const json = rendererer.create(<Icon icon='wechat'/>).toJSON()
        expect(json).toMatchSnapshot()
    })

    it('click successfully', () => {
        const fn = jest.fn()
        const component = mount(<Icon icon='wechat' onClick={fn}/>)
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()
    })

})

