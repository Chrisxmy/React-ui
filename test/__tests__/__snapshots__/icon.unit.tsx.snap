// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<span>
  <svg
    className="m-icon"
    style={
      Object {
        "fill": undefined,
      }
    }
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;
